src/js/models/_markerable.js

```diff
@@ -35,6 +35,31 @@ export default class Markerable extends LinkedItem {
   }
 
   /**
+   * @param {Marker}
+   * @param {Number} markerOffset The offset relative to the start of the marker
+   *
+   * @return {Number} The offset relative to the start of this section
+   */
+  offsetOfMarker(marker, markerOffset) {
+    if (marker.section !== this) {
+      throw new Error(`Cannot get offsetOfMarker for marker that is not child of this`);
+    }
+    // FIXME it is possible, when we get a cursor position before having finished reparsing,
```

Comment:
There must be a nuance here. I'm not sure when we are asking for the offset of a marker, but against dirty or out of sync post ast. It seems bad to do that, I agree, but the problem seems to lie with asking for cursor position in a marker (beyond just the section) before parsing. Hm.